File: SP_Assignment1/Assignment01-MatrixSlicing/python-matrix-slicing/MatrixSlicer.py
class MatrixSlicer:

    def slice_matrix(array, row_start, row_end, col_start, col_end):
        return [row[col_start:col_end + 1] for row in array[row_start:row_end + 1]]

    def print_matrix(matrix):
        for row in matrix:
            print(row)
            
    def display_matrix(matrix, row_start, row_end, col_start, col_end):
        root = tk.Tk()
        root.title("Sliced Matrix")

        for i in range(len(matrix)):
            for j in range(len(matrix[i])):
                value = matrix[i][j]
                label = tk.Label(root, text=str(value), borderwidth=2, relief="solid", width=6, height=3)
                
                if row_start <= i <= row_end and col_start <= j <= col_end:
                    label.config(bg="yellow", fg="red")  

                label.grid(row=i, column=j, padx=5, pady=5)

        root.mainloop()

File: SP_Assignment1/Assignment01-MatrixSlicing/java-matrix-slicing/src/MatrixSlicer.java
package SoftwareParadigms;

import java.awt.Color;
import java.awt.Dimension;
import java.awt.GridLayout;
import java.util.Arrays;
import javax.swing.BorderFactory;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.SwingConstants;

public class MatrixSlicer {

    public static int[][] sliceMatrix(int[][] array, int rowStart, int rowEnd, int colStart, int colEnd) {
        int rowCount = rowEnd - rowStart + 1;
        int colCount = colEnd - colStart + 1;
        int[][] result = new int[rowCount][colCount];

        for (int i = rowStart; i <= rowEnd; i++) {
            for (int j = colStart; j <= colEnd; j++) {
                result[i - rowStart][j - colStart] = array[i][j];
            }
        }

        return result;
    }
 

    public static void printMatrix(int[][] matrix) {
        for (int i = 0; i < matrix.length; i++) {
        System.out.println(Arrays.toString(matrix[i]));
     }
 }
    
    public static void displayMatrix(int[][] matrix, int rowStart, int rowEnd, int colStart, int colEnd) {
        JFrame frame = new JFrame("Sliced Matrix");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setSize(400, 300);
        frame.setLayout(new GridLayout(matrix.length, matrix[0].length));

        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[i].length; j++) {
                JLabel label = new JLabel(String.valueOf(matrix[i][j]), SwingConstants.CENTER);
                
                if (i >= rowStart && i <= rowEnd && j >= colStart && j <= colEnd) {
                    label.setBorder(BorderFactory.createLineBorder(Color.RED, 2)); 
                    label.setBackground(Color.YELLOW); 
                    label.setOpaque(true); 
                }
                
                label.setPreferredSize(new Dimension(50, 30)); 
                frame.add(label);
            }
        }

        frame.pack(); 
        frame.setVisible(true);
    }


   
}
